easypost/__init__.py

```diff
@@ -328,12 +330,14 @@ def requests_request(self, method, abs_url, headers, params):
                 abs_url = self.build_url(abs_url, params)
             data = None
         elif method == 'post' or method == 'put':
-            data = self.encode(params)
+            if headers['Content-Type'] != 'application/json':
+                data = self.encode(params)
         else:
             raise Error("Bug discovered: invalid request method: %s. "
                         "Please report to contact@easypost.com." % method)
-
         try:
+            if headers['Content-Type'] == 'application/json':
```

Comment:
Could probably move this up and merge with lines 333/334 so that you have an if/else.
```
if application/json:
   do these abnormal things
else:
   do the normal things
```
If the json.dumps truly needs to be try/except-wrapped (not sure if it does), you'll probably want to catch whatever exception that it throws so that debugging is easier.